bimap.go

```diff
@@ -4,20 +4,29 @@ package bimap
 import "sync"
 
 // BiMap is a bi-directional hashmap that is thread safe and supports immutability
-type BiMap struct {
+type BiMap[K comparable, V comparable] struct {
 	s         sync.RWMutex
 	immutable bool
-	forward   map[interface{}]interface{}
-	inverse   map[interface{}]interface{}
+	forward   map[K]V
+	inverse   map[V]K
 }
 
 // NewBiMap returns a an empty, mutable, biMap
-func NewBiMap() *BiMap {
-	return &BiMap{forward: make(map[interface{}]interface{}), inverse: make(map[interface{}]interface{}), immutable: false}
+func NewBiMap[K comparable, V comparable]() *BiMap[K, V] {
+	return &BiMap[K, V]{forward: make(map[K]V), inverse: make(map[V]K), immutable: false}
+}
+
+// NewBiMapFrom returns a new BiMap from a map[K, V]
+func NewBiMapFromMap[K comparable, V comparable](forwardMap map[K]V) *BiMap[K, V] {
```

Comment:
could you rebase? I recently merged https://github.com/vishalkuo/bimap/pull/4 which does something similar. Let's just edit that function

Reply:
No problem, Will do it now.